pandas/core/indexes/category.py

```diff
@@ -223,37 +224,10 @@ def _create_from_codes(self, codes, dtype=None, name=None):
         return CategoricalIndex(cat, name=name)
 
     @classmethod
-    def _create_categorical(cls, data, dtype=None):
-        """
-        *this is an internal non-public method*
-
-        create the correct categorical from data and the properties
-
-        Parameters
-        ----------
-        data : data for new Categorical
-        dtype : CategoricalDtype, defaults to existing
-
-        Returns
-        -------
-        Categorical
-        """
-        if isinstance(data, (cls, ABCSeries)) and is_categorical_dtype(data):
-            data = data.values
-
-        if not isinstance(data, ABCCategorical):
-            return Categorical(data, dtype=dtype)
-
-        if isinstance(dtype, CategoricalDtype) and dtype != data.dtype:
-            # we want to silently ignore dtype='category'
-            data = data._set_dtype(dtype)
-        return data
-
-    @classmethod
-    def _simple_new(cls, values, name=None, dtype=None):
+    def _simple_new(cls, values: Categorical, name=None):
```

Comment:
this would fix-up mypy
```suggestion
    def _simple_new(cls, values: Categorical, name=None, dtype=None):
```
and then assert dtype is None or CategoricalDtype?

Reply:
maybe a docstring/comment to say dtype is for compatibility with Index._simple_new and not used, but if specified and not CategoricalDtype will raise.